cmd/cluster/owner.go

```diff
@@ -0,0 +1,187 @@
+	return &ownerOptions{
+		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
+			Flags: flags,
+		},
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *ownerOptions) complete(cmd *cobra.Command, _ []string) error {
+	err := CompleteValidation(&o.k8sclusterresourcefactory, o.IOStreams)
+	if err != nil {
+		return err
+	}
+
+	o.output = o.GlobalOptions.Output
+
+	return nil
+}
+
+func quickTemplate(templateString, replacement string) (string, error) {
+	t, err := template.New("").Parse(templateString)
+	if err != nil {
+		return "", fmt.Errorf("could not create template: %w", err)
+	}
+	// Buffer for pretty output:
+	buf := new(bytes.Buffer)
+	err = t.Execute(buf, replacement)
+	if err != nil {
+		return "", fmt.Errorf("could not execute template: %w", err)
+	}
+	return buf.String(), nil
+}
+
+func (o *ownerOptions) getUser(connection *sdk.Connection) (*amsv1.Account, error) {
+	// if we need to get the current user
+	if o.user == "" {
+		response, err := connection.AccountsMgmt().V1().CurrentAccount().Get().
+			Send()
+		if err != nil {
+			return nil, fmt.Errorf("Can't send request: %w", err)
+		}
+		return response.Body(), nil
+	}
+
+	query, err := quickTemplate("username like '%{{.}}%' or id like '%{{.}}%' or email like '%{{.}}%'", o.user)
+	if err != nil {
+		return nil, fmt.Errorf("could not create template: %w", err)
+	}
+
+	response, err := connection.AccountsMgmt().V1().Accounts().List().
+		Parameter("search", query).Send()
+	if err != nil {
+		return nil, fmt.Errorf("Can't send request: %w", err)
+	}
+
+	if response.Total() != 1 {
+		return nil, fmt.Errorf("could not find an exact match to the query (got %d matches)", response.Total())
+	}
+
+	firstItem := response.Items().Get(0)
+
+	return firstItem, nil
+}
+func (o *ownerOptions) getOwnedClusters(connection *sdk.Connection, userID string) ([]*amsv1.Subscription, error) {
+
+	query, err := quickTemplate("creator.id = '{{.}}' and status != 'Deprovisioned' and status != 'Archived'", userID)
+	if err != nil {
+		return nil, fmt.Errorf("could not create template: %w", err)
+	}
+	response, err := connection.AccountsMgmt().V1().Subscriptions().List().
+		Parameter("search", query).Send()
+	if err != nil {
+		return nil, fmt.Errorf("could not send request: %w", err)
+	}
+
+	subscriptionsWithExternalID := []*amsv1.Subscription{}
+	response.Items().Each(func(sub *amsv1.Subscription) bool {
+		_, ok := sub.GetExternalClusterID()
+		if !ok {
+			fmt.Printf("cluster %s has no external id, skipping", sub.ID())
+			return true
+		}
+		subscriptionsWithExternalID = append(subscriptionsWithExternalID, sub)
+		return true
+	})
+
+	return subscriptionsWithExternalID, nil
+}
+
+func (o *ownerOptions) run() error {
+	//The ocm
+	token := os.Getenv("OCM_TOKEN")
+	if token == "" {
+		ocmToken, err := utils.GetOCMAccessToken()
+		if err != nil {
+			return fmt.Errorf("OCM token not set. Please configure it using the OCM_TOKEN evnironment variable or the ocm cli")
```

Comment:
Quick question: Do we follow a specific style guide? The Go styleguide suggests that all errors should start with lower case. If we want to follow this Line 148 and Line 156 need a minor change :)

Reply:
"provided OCM token not set"

Reply:
sounds good!

Reply:
We could think about using golangci on the long term: https://github.com/golangci/golangci-lint
It's basically the de facto replacement of golint.

Reply:
I just have no idea how our Jenkins pipelines do work exactly or if it would make sense if we have some pipelines in Github Actions and some in Jenkins... probably we should discuss this with more people.

Reply:
I can explain how our CI works if you wish, but offline (and I think I'll record as it's good stuff even if we only see it once)